### ISSUE TITLE
New analyzer: Report on call to 'HasFlag' on enums not marked with 'FlagsAttribute'

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3528]
Add an analyzer that will report a diagnostic on a call of `HasFlag` method on an enum without the `FlagsAttribute` and suggest direct equality check instead.

```csharp
public class C
{
	public enum MyEnum { A, B, }

	public void Do(MyEnum m)
	{
		m.HasFlag(MyEnum.B); // Report here
	}
}
```

**Comment by Evangelink** [comment]
Fixed by #3557
